app/schemas/federation_entities.json

```diff
@@ -0,0 +1,366 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "id": "https://diasporafoundation.org/schemas/federation_entities-1.0.0.json",
```

Comment:
> An alternative would be to use https://diaspora.github.io/diaspora_federation/schemas.

The bad side of using `github.io` is that we depend on GitHub. It is recommended to place schemas within the domain of control. I know that chances are low that GitHub will fake our schemas or we'll experience problems because of some kind of outage. However GitHub experienced [some kinds of attacks in the past](https://en.wikipedia.org/wiki/Censorship_of_GitHub). I don't know if `github.io` suffered of any of this, but that in theory is something that can affect diaspora as a user. It doesn't mean that we're completely safe when using our own domain, but it's better to not depend on a different organization if we have a possibility.

> Also I'm not sure if we want to use a separate versioning for that, or just re-use the same version as the gem?

What do we do if the gem is updated but schema isn't? If yes then we'll have a few version numbers matching the same schema. If not, then what do we do when we need to make a patch modification to schema, while gem is changing minor or major version (according to semver)? I'm not sure we need to couple the version indexes here, because schemas are not that strictly tied to the gem's code. There are even such situations possible, when we change the schema, but don't touch gem implementation at all (e.g. adding another constraint to schema which already presents in implementation). Maybe we should even place the schemas under a different repository and even them in a separate gem?